pkg/util/errors/errors.go

```diff
@@ -29,10 +29,16 @@ type Aggregate interface {
 // is itself an implementation of the error interface.  If the slice is empty,
 // this returns nil.
 func NewAggregate(errlist []error) Aggregate {
-	if len(errlist) == 0 {
+	errs := []error{}
+	for _, err := range errlist {
+		if err != nil {
```

Comment:
The last place I felt the need for this functionality was [here](https://github.com/kargakis/origin/blob/578d92cc58b827d930aeb59027dceb0273a6320e/pkg/route/generator/generate.go#L166). Generally, this if err != nil pattern is disturbing when you don't have more than one way to handle the error and I'm happy when I can skip it. It may or may not feel Go-like, from one pov even nil is considered a value, OTOH what should one do when they have to handle a errors.New("nil") error? Special case it? Start parsing errors?